scripts/migrate_schema.py

```
@@ -304,13 +304,25 @@ def sync_table_columns(source_conn, dest_conn, table_name):
+                    col_def += f' DEFAULT {col_default}'
+                
                 # Handle nullable
+                # If NOT NULL but no default and table has data, add as nullable first
                 if is_null == 'NO':
-                    col_def += ' NOT NULL'
-
-                # Handle default
-                if col_default and 'nextval' not in col_default.lower():
-                    col_def += f' DEFAULT {col_default}'
+                    # Check if table has existing rows
+                    dest_cursor.execute(f'SELECT COUNT(*) FROM "{table_name}";')
```

Ruff (0.14.5)

Possible SQL injection vector through string-based query construction

(S608)